documents/converters.go

```diff
@@ -195,7 +200,13 @@ func toProtocolAttributes(attrs map[AttrKey]Attribute) (pattrs []*coredocumentpb
 		case AttrBytes:
 			pattr.Value = &coredocumentpb.Attribute_ByteVal{ByteVal: attr.Value.Bytes}
 		case AttrTimestamp:
-			pattr.Value = &coredocumentpb.Attribute_TimeVal{TimeVal: attr.Value.Timestamp}
+			buf := new(bytes.Buffer)
+			err := binary.Write(buf, binary.BigEndian, attr.Value.Timestamp.Seconds)
+			if err != nil {
+				return nil, err
+			}
+			b := append(make([]byte, maxTimeByteLength-len(buf.Bytes())), buf.Bytes()...)
```

Comment:
this check wont be necessary once we have a fixed-length byte slice :-D

Reply:
If you supply nanoseconds in json, you're also not guaranteed to have the parser persist them.

Reply:
If you mean through timestamp nano, then go time parser will parse the nano as well